engine/src/main/java/org/terasology/rendering/world/WorldRendererImpl.java

```diff
@@ -319,19 +304,16 @@ private void renderShadowMap() {
     public void renderWorldReflection() {
         PerformanceMonitor.startActivity("Render World (Reflection)");
 
-        DefaultRenderingProcess.getInstance().beginRenderReflectedScene();
-        GL11.glCullFace(GL11.GL_FRONT);
+        LwjglRenderingProcess.getInstance().beginRenderReflectedScene();
         playerCamera.setReflected(true);
 
-        playerCamera.lookThroughNormalized();
+        playerCamera.lookThroughNormalized(); // we don't want the reflected scene to be bobbing or moving with the player
         backdropRenderer.render(playerCamera);
         playerCamera.lookThrough();
 
         chunkShader.activateFeature(ShaderProgramFeature.FEATURE_USE_FORWARD_LIGHTING);
 
         if (renderingConfig.isReflectiveWater()) {
-            GL11.glEnable(GL_LIGHT0);
```

Comment:
I looked into this a little bit more. I thought about re-instating the GL_LIGHT0 but I then realized that it is only used in the rendering pass seen in water reflections. I cannot find an analog opengl light used in the main rendering pass, the opaque scene or anywhere else. So, at this stage I'm thinking of leaving this out and see if somebody complains. 

Concerning the issues you mention Cervator, I'll see what I can do.